spec/bolt/plugin/puppetdb_spec.rb

```diff
@@ -52,14 +52,39 @@
     context "with a name configured" do
       let(:opts) do
         { "query" => '',
-          "name" => 'facts.name_fact' }
+          "target_mapping" => { "name" => 'facts.name_fact' } }
       end
 
       it "sets the uri to the specified name" do
         expect(plugin.resolve_reference(opts))
           .to eq([{ "name" => "thefakeslimcertname" }])
       end
     end
+
+    context "with misplaced config keys" do
+      let(:opts) do
+        { "query" => '',
+          "name" => 'facts.name_fact' }
+      end
+
+      it "raises a validation error" do
+        expect { plugin.resolve_reference(opts) }
+          .to raise_error(/PuppetDB plugin expects keys.*name/m)
+      end
+    end
+
+    context "with unknown keys" do
+      let(:opts) do
+        { "query" => '',
+          "target_mapping" => {},
+          "bad_key" => '' }
+      end
+
+      it "raises a validation error" do
+        expect { plugin.resolve_reference(opts) }
+          .to raise_error(/Unknown keys in PuppetDB plugin:.*bad_key/m)
```

Comment:
FWIW I don't think you need the `.*` here since the output should be static and doesn't look to be very long:
```
.to raise_error(/Unknown keys in PuppetDB plugin: \["bad_key"\]/)
```